src/components/pagination/index.tsx

```diff
@@ -0,0 +1,270 @@
+				return generateTruncatedList(rawitems, currentPage)
+			case 'compact':
+			default:
+				return []
+		}
+	})(type)
+
+	return (
+		<nav className={s.nav} aria-label="Pagination navigation">
+			<ButtonArrow type={type} direction="prev" />
+			{type !== 'compact' && (
+				<ul className={s['page-list']}>
+					{items.map((e, i) => {
+						if (e === 'ellipsis') {
+							// This key will be different on every re-render which is OK
+							const key = `${e}-${i}`
+							return (
+								<li className={classNames(s['page-item'])} key={key}>
+									<div className={s.ellipsis}>...</div>
+								</li>
+							)
+						}
+
+						const isSelected = e === currentPage
+						return (
+							<li
+								className={classNames(s['page-item'], {
+									[s['is-selected']]: isSelected,
+								})}
+								// This key will be stable between re-renders
+								// which prevents react from removing focus from
+								// the previous `document.activeElement`
+								key={e}
+							>
+								<ButtonNumber page={e}>{e}</ButtonNumber>
+							</li>
+						)
+					})}
+				</ul>
+			)}
+			<ButtonArrow type={type} direction="next" />
+		</nav>
+	)
+}
+
+const ButtonArrow = ({ type, direction }: ButtonArrowProps) => {
+	const pagination = usePagination()
+	const handleClick = {
+		next: () => {
+			// clamp upper bound
+			const nextIdx = Math.min(pagination.page + 1, pagination.totalPages)
+			pagination.setPage(nextIdx)
+			pagination.onPageChange(nextIdx)
+		},
+		prev: () => {
+			// clamp lower bound (0)
+			const prevIdx = Math.max(pagination.page - 1, 0)
+			pagination.setPage(prevIdx)
+			pagination.onPageChange(prevIdx)
+		},
+	}[direction]
+
+	const icon = {
+		next: <IconChevronRight16 />,
+		prev: <IconChevronLeft16 />,
+	}[direction]
+
+	const ariaLabel = {
+		next: 'Next page',
+		prev: 'Previous page',
+	}[direction]
+
+	const label = {
+		next: 'Next',
+		prev: 'Previous',
+	}[direction]
+
+	const isDisabled = {
+		next: pagination.page === pagination.totalPages,
+		prev: pagination.page === 1,
+	}[direction]
+
+	return (
+		<button
+			className={classNames(
+				s.arrow,
+				s.control,
+				{
+					next: s.next,
+					prev: s.prev,
+				}[direction]
+			)}
+			aria-label={ariaLabel}
+			onClick={handleClick}
+			disabled={isDisabled}
+		>
+			{icon}
+			{type === 'compact' && (
+				<span className={classNames(s.label)} aria-hidden="true">
+					{label}
+				</span>
+			)}
+		</button>
+	)
+}
+
+const ButtonNumber = ({
+	children,
+	page,
+}: PropsWithChildren<ButtonNumberProps>) => {
+	const pagination = usePagination()
+	const handleClick = () => {
+		pagination.setPage(page)
+		// don't trigger callback if the current page is already selected
+		if (pagination.page !== page) {
+			pagination.onPageChange(page)
+		}
+	}
+	return (
+		<button
+			className={classNames(s.control)}
+			type="button"
+			onClick={handleClick}
+		>
+			<span className={'g-screen-reader-only'}>page&nbsp;</span>
+			{children}
+		</button>
+	)
+}
```

Comment:
🚧 Two things here:

- The `children` and `page` props are the same, so both props aren't needed. I'd vote for keeping just `page` since it's more descriptive in the `handleClick` function.
- The `g-screen-reader-only` `span` is redundant because `button` automatically handles setting its Accessible Name we want using its children.

```tsx
const ButtonNumber = ({ page }: ButtonNumberProps) => {
	const pagination = usePagination()
	const handleClick = () => {
		pagination.setPage(page)
		// don't trigger callback if the current page is already selected
		if (pagination.page !== page) {
			pagination.onPageChange(page)
		}
	}
	return (
		<button
			className={classNames(s.control)}
			type="button"
			onClick={handleClick}
		>
			{page}
		</button>
	)
}
```

Reply:
Handled in https://github.com/hashicorp/dev-portal/pull/1508/commits/a1da1faafd4ad8fa6570bff9afca0255e9351a12